ext/reflection/php_reflection.c

```diff
@@ -3699,6 +3699,12 @@ ZEND_METHOD(ReflectionClassConstant, isProtected)
 }
 /* }}} */
 
+/* Returns whether this constant is final */
+ZEND_METHOD(ReflectionClassConstant, isFinal)
+{
+	_class_constant_check_flag(INTERNAL_FUNCTION_PARAM_PASSTHRU, ZEND_ACC_FINAL);
```

Comment:
You also need to add it to the getModifiers() mask and add an IS_FINAL constant to the class.